profile/src/test/java/com/scottlogic/datahelix/generator/profile/AtomicConstraintDeserialiserTests.java

```diff
@@ -110,7 +111,7 @@ public void shouldDeserialiseInSetCsvFileWithoutException() throws IOException {
         // Assert
         InSetConstraintDTO expected = new InSetConstraintDTO();
         expected.field = "country";
-        expected.values = Collections.singletonList("test");
+        expected.values = Collections.singletonList(new WeightedElement<>("test", 1.0));
```

Comment:
Are there any additional tests we should add to this PR - beyond fixing this test.

There are some new scenarios that are catered for, i think there should be tests for these scenarios added here?

Reply:
I definitely agree that there should be some more tests, I just wasn't sure what kind of tests I'd need to write. It's going to be difficult to do an e2e test and see if it's roughly the right distribution. So I assume we'd need a few unit tests in various places? I'm not very familiar with the code base, do you have any suggestions?

Reply:
I've added a few tests that should cover the main parts of these changes.